gateway-ha/src/main/java/io/trino/gateway/ha/handler/RoutingDestination.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.handler;
+
+import java.net.URI;
+
+public record RoutingDestination(String routingGroup, String clusterHost, URI clusterUri) {}
```

Comment:
Are all parameters nullable? Please add a compact constructor to verify non-null parameters.